From 02a9c9ee467c3674e216cc3fd2d64cad6c8ee42b Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 17 Jan 2024 18:17:34 +0100
Subject: [PATCH 480/954] refactor(stm32mp2): remove firewall file

The functions in stm32mp2_fconf_firewall.c do nothing and the memory
firewall configuration that were done in stm32mp1_fconf_firewall.c for
STM32MP1x platform are now managed in RISAF driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b04831daa6fa61fa06cc2fc59f1044133d83251d refactor(stm32mp2): remove firewall file]
Change-Id: I81bf8d6f94e30afe2135538a916de5102bd503f3
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/352186
Domain-Review: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Tested-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Reviewed-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c           |  3 ---
 plat/st/stm32mp2/include/stm32mp2_private.h |  5 +---
 plat/st/stm32mp2/platform.mk                |  3 +--
 plat/st/stm32mp2/stm32mp2_fconf_firewall.c  | 26 ---------------------
 4 files changed, 2 insertions(+), 35 deletions(-)
 delete mode 100644 plat/st/stm32mp2/stm32mp2_fconf_firewall.c

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index 7f238b8cf3..3ba61a2a3d 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -378,8 +378,6 @@ skip_console_init:
 		initialize_pmic();
 	}
 
-	stm32mp2_arch_security_setup();
-
 	fconf_populate("TB_FW", STM32MP_DTB_BASE);
 
 #if STM32MP_DDR_FIP_IO_STORAGE || TRUSTED_BOARD_BOOT
@@ -682,7 +680,6 @@ void bl2_el3_plat_prepare_exit(void)
 	flush_dcache_range(DATA_START, DATA_END - DATA_START);
 
 	stm32mp_io_exit();
-	stm32mp2_security_setup();
 
 	/* Unmask potential tamper before exit */
 	stm32mp_syscfg_mask_potential_tamper_disable();
diff --git a/plat/st/stm32mp2/include/stm32mp2_private.h b/plat/st/stm32mp2/include/stm32mp2_private.h
index 7e69b46090..30dc6656b2 100644
--- a/plat/st/stm32mp2/include/stm32mp2_private.h
+++ b/plat/st/stm32mp2/include/stm32mp2_private.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2023-2024, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -9,9 +9,6 @@
 
 void configure_mmu(void);
 
-void stm32mp2_arch_security_setup(void);
-void stm32mp2_security_setup(void);
-
 uint32_t stm32mp_syscfg_get_chip_version(void);
 uint32_t stm32mp_syscfg_get_chip_dev_id(void);
 void stm32mp_syscfg_mask_potential_tamper_enable(void);
diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 98be75bf13..bafb210128 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -206,8 +206,7 @@ PLAT_BL_COMMON_SOURCES	+=	drivers/st/clk/clk-stm32-core.c				\
 
 PLAT_BL_COMMON_SOURCES  	+=	drivers/st/nvmem/stm32mp_tamp_nvram_mp2.c
 
-BL2_SOURCES			+=	plat/st/stm32mp2/plat_bl2_mem_params_desc.c		\
-					plat/st/stm32mp2/stm32mp2_fconf_firewall.c
+BL2_SOURCES		+=	plat/st/stm32mp2/plat_bl2_mem_params_desc.c
 
 BL2_SOURCES			+=	drivers/st/crypto/stm32_hash.c				\
 					plat/st/stm32mp2/bl2_plat_setup.c			\
diff --git a/plat/st/stm32mp2/stm32mp2_fconf_firewall.c b/plat/st/stm32mp2/stm32mp2_fconf_firewall.c
deleted file mode 100644
index 0d41c3a5f3..0000000000
--- a/plat/st/stm32mp2/stm32mp2_fconf_firewall.c
+++ /dev/null
@@ -1,26 +0,0 @@
-/*
- * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
- *
- * SPDX-License-Identifier: BSD-3-Clause
- */
-
-#include <platform_def.h>
-
-#include <lib/fconf/fconf.h>
-#include <lib/object_pool.h>
-#include <tools_share/firmware_image_package.h>
-
-void stm32mp2_arch_security_setup(void)
-{
-}
-
-void stm32mp2_security_setup(void)
-{
-}
-
-static int fconf_populate_stm32mp2_firewall(uintptr_t config)
-{
-	return 0;
-}
-
-FCONF_REGISTER_POPULATOR(FW_CONFIG, stm32mp2_firewall, fconf_populate_stm32mp2_firewall);
-- 
2.53.0

